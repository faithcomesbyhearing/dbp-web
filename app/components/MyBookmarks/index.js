/**
 *
 * MyBookmarks
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import Link from 'next/link';
import SvgWrapper from '../SvgWrapper';
import LegacyLink from '../LegacyLink';

class MyBookmarks extends React.PureComponent {
	render() {
		const {
			bookmarks,
			deleteNote,
			getNoteReference,
			toggleNotesModal,
			getFormattedNoteDate,
		} = this.props;
		return bookmarks.map((listItem) => (
			<div key={listItem.id} id={listItem.id} className={'highlight-item'}>
				<LegacyLink
					as={`/bible/${listItem.bible_id}/${listItem.book_id}/${
						listItem.chapter
					}`}
					href={`/bible/${listItem.bible_id}/${listItem.book_id}/${
						listItem.chapter
					}`}
				>
					<button
						onClick={toggleNotesModal}
						className="list-item"
						type="button"
					>
						<div className="title-text">
							<h4 className="title">
								<span className="date">
									{getFormattedNoteDate(listItem.created_at)}
								</span>{' '}
								| {getNoteReference(listItem, true)}
							</h4>
							<p className="text">{listItem.bible_id}</p>
						</div>
					</button>
				</LegacyLink>
				<button
					key={`${listItem.id}-delete`}
					onClick={() => deleteNote({ noteId: listItem.id, isBookmark: true })}
					className={'delete-highlight'}
					tabIndex={0}
					type="button"
				>
					<SvgWrapper className={'icon'} svgid={'delete'} />
					<span>Delete</span>
				</button>
			</div>
		));
	}
}

MyBookmarks.propTypes = {
	bookmarks: PropTypes.array,
	getFormattedNoteDate: PropTypes.func,
	getNoteReference: PropTypes.func,
	toggleNotesModal: PropTypes.func,
	deleteNote: PropTypes.func,
};

export default MyBookmarks;
