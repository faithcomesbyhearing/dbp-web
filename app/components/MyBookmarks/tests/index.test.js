import React from 'react';
import { render } from '@testing-library/react';
import MyBookmarks from '..';
import {
	getFormattedNoteDate,
	getNoteReference,
	bookmarks,
} from '../../../utils/testUtils/notebookFunctions';

const props = {
	bookmarks,
	getFormattedNoteDate: jest.fn((params) => getFormattedNoteDate(params)),
	getNoteReference: jest.fn((params) => getNoteReference(params)),
	toggleNotesModal: jest.fn(),
	deleteNote: jest.fn(),
};

describe('<MyBookmarks /> component', () => {
	it('should match snapshot with expected props', () => {
		const { asFragment } = render(<MyBookmarks {...props} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
