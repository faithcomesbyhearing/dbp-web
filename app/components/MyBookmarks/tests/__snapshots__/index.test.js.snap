// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyBookmarks /> component should match snapshot with expected props 1`] = `
[
  <div
    className="highlight-item"
    id={5452296}
  >
    <a
      className="list-item"
      href="/bible/ENGESV/MAT/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="title-text"
      >
        <h4
          className="title"
        >
          <span
            className="date"
          >
            01.18.19
          </span>
           
          | 
          Matthew 1:undefined
        </h4>
        <p
          className="text"
        >
          ENGESV
        </p>
      </div>
    </a>
    <div
      className="delete-highlight"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        className="icon"
      >
        <use
          xlinkHref="/svglist.svg#delete"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <span>
        Delete
      </span>
    </div>
  </div>,
  <div
    className="highlight-item"
    id={5452373}
  >
    <a
      className="list-item"
      href="/bible/ENGESV/MAT/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="title-text"
      >
        <h4
          className="title"
        >
          <span
            className="date"
          >
            01.18.19
          </span>
           
          | 
          Matthew 1:undefined
        </h4>
        <p
          className="text"
        >
          ENGESV
        </p>
      </div>
    </a>
    <div
      className="delete-highlight"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        className="icon"
      >
        <use
          xlinkHref="/svglist.svg#delete"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <span>
        Delete
      </span>
    </div>
  </div>,
  <div
    className="highlight-item"
    id={5574788}
  >
    <a
      className="list-item"
      href="/bible/ENGESV/MAT/3"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="title-text"
      >
        <h4
          className="title"
        >
          <span
            className="date"
          >
            02.06.19
          </span>
           
          | 
          Matthew 3:3
        </h4>
        <p
          className="text"
        >
          ENGESV
        </p>
      </div>
    </a>
    <div
      className="delete-highlight"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        className="icon"
      >
        <use
          xlinkHref="/svglist.svg#delete"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <span>
        Delete
      </span>
    </div>
  </div>,
  <div
    className="highlight-item"
    id={5574789}
  >
    <a
      className="list-item"
      href="/bible/ENGESV/MRK/16"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="title-text"
      >
        <h4
          className="title"
        >
          <span
            className="date"
          >
            02.08.19
          </span>
           
          | 
          Mark 16:3
        </h4>
        <p
          className="text"
        >
          ENGESV
        </p>
      </div>
    </a>
    <div
      className="delete-highlight"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        className="icon"
      >
        <use
          xlinkHref="/svglist.svg#delete"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <span>
        Delete
      </span>
    </div>
  </div>,
  <div
    className="highlight-item"
    id={5574790}
  >
    <a
      className="list-item"
      href="/bible/ENGNIV/MAT/11"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="title-text"
      >
        <h4
          className="title"
        >
          <span
            className="date"
          >
            02.12.19
          </span>
           
          | 
          Matthew 11:7
        </h4>
        <p
          className="text"
        >
          ENGNIV
        </p>
      </div>
    </a>
    <div
      className="delete-highlight"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        className="icon"
      >
        <use
          xlinkHref="/svglist.svg#delete"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <span>
        Delete
      </span>
    </div>
  </div>,
]
`;
